sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2011,7 +2011,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
 
   def batchMetadataFileNotFoundError(batchMetadataFile: Path): SparkFileNotFoundException = {
     new SparkFileNotFoundException(
-      errorClass = "_LEGACY_ERROR_TEMP_2206",
+      errorClass = "CANNOT_FIND_BATCH",
```

Comment:
I understand we want error class to be not too verbose, but CANNOT_FIND_BATCH sounds to me misleading - no one knows that's due to missed file.

Maybe, CANNOT_FIND_BATCH_METADATA_FILE, or CANNOT_FIND_BATCH_METADATA?

Reply:
ok, let's follow our conventions for not found objects:
- ENCODER_NOT_FOUND
- FIELD_NOT_FOUND
- INDEX_NOT_FOUND
- PARTITIONS_NOT_FOUND

I would propose `BATCH_METADATA_NOT_FOUND`

Reply:
BATCH_METADATA_NOT_FOUND sounds OK to me. Thanks for suggesting good alternatives!